mlir/test/Transforms/remove-dead-values.mlir

```diff
@@ -3,9 +3,10 @@
 // The IR is updated regardless of memref.global private constant
 //
 module {
-  memref.global "private" constant @__something_global : memref<i32> = dense<0>
+  memref.global "private" constant @global_buffer : memref<5xi32> = dense<[1, 2, 3, 4, 5]> : tensor<5xi32>
   func.func @main(%arg0: i32) -> i32 {
     %0 = tensor.empty() : tensor<10xbf16>
+    %1 = memref.get_global @global_buffer : memref<5xi32>
```

Comment:
I think we should CHECK in the output what we expect?

Reply:
@joker-eph We are checking the output, below the output after running this test but I have added and extra FILECHECK for `memref.global`

```
module {
  memref.global "private" constant @__constant_4xi32 : memref<4xi32> = dense<[1, 2, 3, 4]> {alignment = 16 : i64}
  func.func @main(%arg0: i32) -> i32 {
    return %arg0 : i32
  }
}
```

Reply:
The existing test is explicitly checking that the `tensor.empty` op is removed:
```
    // CHECK-NOT: tensor.empty
```

Right now you're not testing the expectation that the get_global op is removed, so I would add:

```
    // CHECK-NOT: memref.get_global
```

Reply:
Thank you for the clarification, I have updated the test with the required FILECHECK.